scipy/stats/_distribution_infrastructure.py

```diff
@@ -3514,7 +3517,43 @@ def make_distribution(dist):
     Parameters
     ----------
     dist : `rv_continuous`
-        Instance of `rv_continuous`.
+        Instance of `rv_continuous` OR an instance of any class with the following
+        attributes:
+
+        __make_distribution_version__ : str
+            A string containing the version number of SciPy in which this interface
+            is defined. The preferred interface may change in future SciPy versions,
+            in which case support for an old interface version may be deprecated
+            and eventually removed.
+        parameters : dict
+            A dictionary describing the parameters of the distribution.
+            Each key is the name of a parameter,
+            and the corresponding value is itself a dictionary with the following items.
+
+            endpoints : tuple
+                A tuple defining the lower and upper endpoints of the domain of the
+                parameter. Allowable values are floats or the name (string) of another
+                parameter.
+
+            inclusive : tuple of bool
+                A tuple specifying whether the endpoints are included within the domain
+                of the parameter.
+
+        support : tuple
+            A tuple defining the lower and upper endpoints of the support of the
+            distribution. Allowable values are floats or the name (string) of a
+            parameter.
+
+        The class **must** also define a ``pdf`` method and **may** define methods
+        ``logentropy``, ``entropy``, ``median``, ``mode``, ``logpdf``,
+        ``logcdf``, ``cdf``, ``logccdf``, ``ccdf``,
+        ``ilogcdf``, ``icdf``, ``ilogccdf``, and ``iccdf````.
+        All methods must accept the parameters of the distributions as keyword arguments
+        and any positional-only arguments accepted by the corresponding method of
+        `ContinuousDistribution`. Methods ``moment_raw``, ``moment_central``,
+        ``moment_standardized`` may be defined to accept the ``order`` of the moment
+        by position, acccept all distribution parameters by keyword, and return the raw,
+        central, and standardized moments of the distribution, respectively.
```

Comment:
```suggestion
        If defined, these methods must accept the parameters of the distributions as
        keyword arguments and also accept any positional-only arguments accepted by
        the corresponding method of `ContinuousDistribution`. Methods ``moment_raw``,
        ``moment_central``, ``moment_standardized`` may also be optionally defined, and
         if so, must accept the ``order`` of the moment by position, accept all distribution
        parameters by keyword, and return the raw, central, and standardized moments
        of the distribution, respectively.
```
I'm not sure if it's post-flu brain fog, but I had some trouble parsing this out. The suggestion is my attempt to disambiguate based on my understanding of what was meant. If you think this looks good, I'd advise against directly committing the suggestion, because I typed it directly into the GitHub comment box, and it may not be formatted correctly.

Reply:
This was a lot of information that I was trying to express compactly, and perhaps I went a bit too far. Sure, we can decompress it a bit.

Reply:
```suggestion
        If defined, these methods must accept the parameters of the distributions as
        keyword arguments and also accept any positional-only arguments accepted by
        the corresponding method of `ContinuousDistribution`. Methods ``moment_raw``,
        ``moment_central``, ``moment_standardized`` may also be defined; if so,
        they must accept the ``order`` of the moment by position, accept all
        distribution parameters by keyword, and return the raw, central, and
        standardized moments of the distribution, respectively.
```